Add Premium badge to premium themes in the design preview header

mreishus opened the pull request:
#### Proposed Changes

* Add Premium badge to premium themes in the design preview header

![2022-07-25_16-46](https://user-images.githubusercontent.com/937354/180882800-b0d55b40-3e5e-48e8-847d-a70ff2b26ad0.png)


#### Testing Instructions

* Visit `http://calypso.localhost:3000/setup/designSetup?siteSlug=YOURSITE.wordpress.com&flags=-signup/design-picker-unified`
* Also visit `http://calypso.localhost:3000/setup/designSetup?siteSlug=YOURSITE.wordpress.com&flags=signup/design-picker-unified`
* Click on a premium theme, on the preview page, observe the header and look for the premium badge
* Go back, repeat for a non-premium theme, the badge should be missing

----


#### Issue with mouseover

If you mouseover the badge, the tooltip does not appear. This is related to the way the steps are implemented in stepper. The steps are in an absolute positioned div, where the middle of it is transparent:

![2022-07-25_16-59](https://user-images.githubusercontent.com/937354/180883299-2400e61e-6077-4d68-ab6e-fc874b5757ad.png)

If you try to select text in the title, you cannot, because the absolute positioned div is blocking the title under it:

https://user-images.githubusercontent.com/937354/180883364-83d0d333-e262-4fbe-8751-1e160e47f9b9.mp4

The same mechanism is blocking the mouseover for the tooltip in this change.
I have a potential fix, but I don't know if it will break anything. I also wasn't sure if I should put it in this PR, or another PR, or if we should not do it. (Update: Created PR #65971)

( See #65971 for the diff )

This blocks mouse events on the absolutely positioned div and restores them to the buttons on either side of it that are filled in. Once again, not sure if it works completely because I'm pretty new to stepper.



Related to #65786

mreishus committed:
Add Premium badge to premium themes in the design preview header

blackjackkent commented:
> I also wasn't sure if I should put it in this PR, or another PR, or if we should not do it

Those further changes at a glance don't seem like they should cause stepper related problems since it's just a style related change? (Unless I'm missing some context here.) I do think a separate PR makes sense though.

mreishus commented:
Split it out into #65971

andres-blanco commented:
There are two differences comparing with the Figma file (0EogtxvlmxMGcnEEaMVECv-fi-17%3A415) 
* The badge is vertically centered with respect to the title.
* The font of the badge is different from the one in the title.

| Figma  | This PR |
| ------------- | ------------- |
| <img width="427" alt="image" src="https://user-images.githubusercontent.com/375980/181081345-ea0efe04-1cbe-445c-b27c-b83fe8485621.png"> | <img width="265" alt="image" src="https://user-images.githubusercontent.com/375980/181081428-1d6231af-7d1b-402f-b112-77b8a7055264.png"> |

mreishus commented:
Also noting that after the design picker changes in #65534, these changes are no longer appearing in localhost.

* `client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/site-setup-design-picker.tsx` Changed in this PR, used when flag (`signup/design-picker-unified`) is off
* `client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx` Used when flag is on

Currently prod has flag off, and dev has flag on. 
This PR will need to change both versions, and possibly extract the logic/layout used to one file between the two.

mreishus committed:
Extract to DesignPickerDesignTitle

Changes will now appear on both unified and non-unified versions

mreishus commented:
Updated so the changes will appear with both the unified flag on and flag off. (Also added flag specific urls to the original description).
Still need to look at alignment and fonts.

mreishus committed:
Fix alignment

mreishus committed:
Force premium-badge font to be sans

mreishus commented:
Ok, think I fixed the alignment and font issues.

andres-blanco reviewed (approved):
Tested with the `signup/design-picker-unified` flag on & off and it works fine.
